core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -177,15 +212,18 @@ public void createTableUnsafe(
             TableStructure struct
     ) {
         securityContext.checkWritePermission();
-        tableRegistry.registerTable(struct);
+        int tableId = (int) tableIdGenerator.getNextId();
```

Comment:
perhaps because it's already saved as long in the files

Reply:
I guess it's not the case for WAL tables at least. The unpleasant side effect of downcast is having duplicates as soon as the counter reaches `Integer.MAX_VALUE`. On the other hand, in case of an `int` counter we'd get negative values which is also unpleasant.